selfupdate.go

```diff
@@ -17,18 +18,64 @@ type updateRequest struct {
 	SBOMHash  string `json:"sbom_hash"`
 }
 
+type attemptRequest struct {
+	MachineID string `json:"machine_id"`
+	SBOMHash  string `json:"sbom_hash"`
+}
+
 type updateResponse struct {
 	SBOMHash     string `json:"sbom_hash"`
 	RegistryType string `json:"registry_type"`
 	Link         string `json:"download_link"`
 }
 
-func checkForUpdates(ctx context.Context, updateEndpoint string, request *updateRequest) (*updateResponse, error) {
+const (
+	updateAPI        = "api/v1/update"
+	attemptUpdateAPI = "api/v1/attempt"
+)
+
+func registerUpdateAttempt(ctx context.Context, gusServer string, request *attemptRequest) error {
```

Comment:
Should we take this opportunity to move both requests to the new generated API client? https://pkg.go.dev/github.com/gokrazy/gokapi/gusapi#UpdateApiService.Attempt

Reply:
Did this in a follow up PR, here we go: https://github.com/gokrazy/selfupdate/pull/3